c/driver/flightsql/meson.build

```diff
@@ -47,6 +47,13 @@ adbc_driver_flightsql_lib = custom_target(
     install_dir : '.',
 )
 
+pkg.generate(
+    name: 'adbc_driver_flightsql',
+    description: 'ADBC Driver for FlightSQL',
```

Comment:
Could you use the same content as existing `.pc`?
https://github.com/apache/arrow-adbc/blob/main/c/driver/flightsql/adbc-driver-flightsql.pc.in
(If the existing content isn't strange, could you improve it?)

Reply:
Sounds good. I noticed two of the CMake config files were missing the URL so added those as well. Happy to split off if you want to manage that in a separate PR